### PR TITLE
Patch dask-expr `var` logic in dask-cudf

### PR DESCRIPTION
**Pull request by rjzamora**
## Description
The `var` logic in dask-expr relies on pandas -> numpy conversion that does not work for cudf -> cupy when null values are present. This PR copies over the custom `var` logic being used in dask-cudf for the legacy API.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — CHANGES_REQUESTED** [review]
We need to add a test in dask-cudf for the functions used in the 10min notebook.

**Comment by rjzamora** [comment]
>We need to add a test in dask-cudf for the functions used in the 10min notebook.

Agreed - This is still a draft.

File: python/dask_cudf/dask_cudf/expr/__init__.py [hunk 1]
```diff
@@ -7,12 +7,12 @@
 QUERY_PLANNING_ON = config.get("dataframe.query-planning", None) is not False
 
 # Register custom expressions and collections
-try:
-    import dask_cudf.expr._collection
-    import dask_cudf.expr._expr
+if QUERY_PLANNING_ON:
```
[the remark this answers is not in the document]
Comment (bdice) [reply python/dask_cudf/dask_cudf/expr/__init__.py:10]: Merge the upstream to fix this.

**Comment by rjzamora** [comment]
CI failures are due to a bad/old dask-expr version being pulled (https://github.com/dask/distributed/issues/8574)

**Comment by rjzamora** [comment]
The dask/dask-expr nightly-conda situation should be fixed now. So dask-related CI *should* start passing now.

**Comment by bdice** [comment]
I reran the failed CI jobs. @rjzamora Is this ready for review?

File: python/dask_cudf/dask_cudf/expr/_expr.py [hunk 2]
```diff
@@ -25,19 +29,95 @@ def _kwargs(self) -> dict:
...
-_dx_reduction_aggregate = Var.reduction_aggregate
+# The upstream Var code uses `Series.values`, and relies on numpy
+# for most of the logic. Unfortunately, cudf -> cupy conversion
+# is not supported for data containing null values. Therefore,
+# we must implement our own version of Var for now. This logic
+# is mostly copied from dask-cudf.
+
+
+class VarCudf(Reduction):
+    # Uses the parallel version of Welford's online algorithm (Chan 79')
```
Comment (bdice) [inline python/dask_cudf/dask_cudf/expr/_expr.py:40]:
```suggestion
    # Uses the parallel version of Welford's online algorithm (Chan '79)
```

**Review by bdice — APPROVED** [review]
Changes look fine to me.

**Comment by rjzamora** [comment]
>Is this ready for review?

Yeah, thanks @bdice !

**Review by raydouglass — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Pull request merged**